### PR TITLE
feat: Add helpful tooltips for the key features

### LINKED ISSUE #1640: Enhance: There are no definitions in the UI yet for key features
As a user, I would like descriptions for key features available in the UI. 

For features and functionality that are relevant to the view I am in that tell me: 

- What is it? 
- What should I know about it? / What does it do for me? 
- What is is required for? (i.e. you have to create a workspace using a template)
- Where do I make changes to it?

For instance:

- Templates 
- Workspaces
- Resource 
- Agent

These could be info icon pop overs, collapsible descriptions, etc.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
- Add PageHeader component
- Add HelpTooltip component

Closes #1640 

Screen Shot 2022-06-06 at 15 11 58
Screen Shot 2022-06-06 at 15 11 45

### CODE REVIEW

File: site/src/components/HelpTooltip/HelpTooltip.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <HelpTooltipTitle>What is template?</HelpTooltipTitle>
+    <HelpTooltipText>
+      With templates you can create a common configuration for your workspaces using Terraform. So, you and your team
+      can use the same environment to deliver great software.
+    </HelpTooltipText>
+    <HelpTooltipLinksGroup>
+      <HelpTooltipLink href="https://github.com/coder/coder/">Creating a template</HelpTooltipLink>
+      <HelpTooltipLink href="https://github.com/coder/coder/">Updating a template</HelpTooltipLink>
+    </HelpTooltipLinksGroup>
+  </HelpTooltip>
```
Comment (BrunoQuaresma) [inline site/src/components/HelpTooltip/HelpTooltip.stories.tsx:27]: cc.: @tjcran @khorne3 can you review the copy here, please?

File: site/src/components/Resources/Resources.tsx [hunk 2]
```diff
@@ -23,6 +30,31 @@ const Language = {
...
+}
+
+const AgentHelpTooltip: React.FC = () => {
+  return (
+    <HelpTooltip size="small">
+      <HelpTooltipTitle>What is an agent?</HelpTooltipTitle>
+      <HelpTooltipText>An agent is a software that executes Coder inside of the resource.</HelpTooltipText>
+    </HelpTooltip>
+  )
+}
```
Comment (BrunoQuaresma) [inline site/src/components/Resources/Resources.tsx:56]: cc.: @tjcran @khorne3 can you review the copy here, please?

File: site/src/pages/WorkspacesPage/WorkspacesPageView.tsx [hunk 3]
```diff
@@ -25,19 +43,147 @@ export const Language = {
...
+        <HelpTooltipLink href="https://github.com/coder/coder/blob/main/docs/workspaces.md#connect-with-ssh">
+          Connect with SSH
+        </HelpTooltipLink>
+        <HelpTooltipLink href="https://github.com/coder/coder/blob/main/docs/workspaces.md#editors-and-ides">
+          Editors and IDEs
+        </HelpTooltipLink>
+      </HelpTooltipLinksGroup>
+    </HelpTooltip>
+  )
+}
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacesPage/WorkspacesPageView.tsx:73]: cc.: @tjcran @khorne3 can you review the copy here, please?

**Review by Kira-Pilot — APPROVED** [review]
I think we could use some `Language` objects for the tooltip copy. I understand the tooltip copy is still be reviewed but once that's refined, this looks great from a FE perspective!

File: site/src/components/HelpTooltip/HelpTooltip.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+const Template: Story<HelpTooltipProps> = (args) => (
+  <HelpTooltip {...args}>
+    <HelpTooltipTitle>What is template?</HelpTooltipTitle>
+    <HelpTooltipText>
+      With templates you can create a common configuration for your workspaces using Terraform. So, you and your team
+      can use the same environment to deliver great software.
+    </HelpTooltipText>
+    <HelpTooltipLinksGroup>
+      <HelpTooltipLink href="https://github.com/coder/coder/">Creating a template</HelpTooltipLink>
+      <HelpTooltipLink href="https://github.com/coder/coder/">Updating a template</HelpTooltipLink>
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/components/HelpTooltip/HelpTooltip.stories.tsx:25]: Ahh, this one is just a sample. I'm already using this link for the templates: 
```
 <HelpTooltipLink href="https://github.com/coder/coder/blob/main/docs/templates.md#manage-templates">
  Manage templates
</HelpTooltipLink>
```

**Pull request merged**